src/client/Client.cc

```diff
@@ -6124,6 +6124,10 @@ int Client::may_open(Inode *in, int flags, const UserPerm& perms)
   int r = 0;
   switch (in->mode & S_IFMT) {
     case S_IFLNK:
+#if defined(__linux__) && defined(O_PATH)
+      if (flags & O_PATH)
+        break;
+#endif
```

Comment:
I have a question - [open(2)](open(2)) mentions of this special case only if the `O_NOFOLLOW` is specified, right? but with the code above it looks like no matter if it's `O_NOFOLLOW` or not, we just move ahead with the symlink and not fail? If this is so, then if this fd (which is a target file and not a symlink fd) runs with empty pathname in `readlinkat` then it works?

Reply:
No. It returns `ENOENT` because the fd itself doesn't point to a symlink as it should have been already resolved during _open()_ without `O_NOFOLLOW`. This is the special case we intercept with the other change inside _readlinkat()_.

Reply:
Got it, thanks @anoopcs9